subsys/net/Kconfig

```diff
@@ -18,8 +18,8 @@ if NET_BUF
 
 config NET_BUF_USER_DATA_SIZE
 	int "Size of user_data available in every network buffer"
-	default 4
-	range 4 65535 if BT
+	default 8
+	range 8 65535 if BT
```

Comment:
This will penalize all net_buf users of the system and increase the runtime memory consumption noticeably if the system is configured with many buffers (this will in particular happen when IP networking support is enabled). So I'd suggest you look for a solution which doesn't require increasing the user data. We've been in this kind of situations many times, where it looked like we needed to increase the user data size, but it turned out to always be possible to find an alternative, simple enough solution which didn't require it. If I remember right, this is one of the reasons `struct bt_conn_tx` is independent from any buffer.

Reply:
I'm assuming that the change from 4 to 8 is when you have 64 bits pointers.
Then this line #define conn_tx(buf) ((struct conn_tx_cb *)net_buf_user_data(buf)) requires 8 bytes.
@jhedberg This line was here before, it was only moved. Any suggestions on how this should be handled?

Reply:
Any update on fixing this user data issue?  It can probably be solved by adding a dedicated user data array along side of acl_tx_pool, in conn.c, and then picking elements from it using `user_data = &acl_tx_user_data[net_buf_id(buf)]`. mesh/adv.c does the same thing since it needs 12 bytes of user data per buffer.

Reply:
@joerchan typed the above message before seeing yours. Is my suggestion with net_buf_id() clear enough?

Reply:
@joerchan also, I don't think this has anything to do with 64-bit pointers since Zephyr doesn't support such architectures (right now at least). In the case of this PR it's a function pointer + 4 bytes of user data passed to that function, where the 8 bytes seems to come from.

Reply:
Ok, if there are no 64-bit pointers then 4 bytes of user data should be enough. The net_buf_id(buf) suggestion is clear, thank you.